XMonad/Layout/Circle.hs

```diff
@@ -15,11 +15,11 @@
 --
 -----------------------------------------------------------------------------
 
-module XMonad.Layout.Circle (
-                             -- * Usage
-                             -- $usage
-                             Circle (..)
-                            ) where -- actually it's an ellipse
+module XMonad.Layout.Circle {-# DEPRECATED "Use XMonad.Layout.CircleEx instead" #-}
```

Comment:
If you export `CircleEx`, then (I think) this whole module could be replaced by something like

``` haskell
pattern Circle :: CircleEx a
pattern Circle = CircleEx 1 (70 % 99) (2 % 5) 1 0
```

In this way we wouldn't duplicate the implementation details

Reply:
That gives me

```
/home/portnov/src/xmonad-contrib-portnov/XMonad/Layout/Circle.hs:43:40: error: Parse error in pattern: 2 % 5
   |
43 | pattern Circle = CircleEx 1 (70 % 99) (2%5) 1 0
   |                                        ^^^
```

I suspect `pattern` thing is only applicable to literals, not expressions.

Reply:
Ah, right. In that case I guess `GHC.Real.(:%)` should work?

Reply:
Try it with the extra spaces; ghc since 9.2 requires spaces around infix operators, as part of rationalizing parsing of `OverloadedRecordDot`, `BangPatterns`, `NumericLiterals`, and various other special forms.